pkg/apis/elasticsearch/v1alpha1/types.go

```diff
@@ -61,6 +61,8 @@ type ElasticsearchNode struct {
 	Resources    v1.ResourceRequirements  `json:"resources"`
 	NodeSelector map[string]string        `json:"nodeSelector,omitempty"`
 	Storage      ElasticsearchStorageSpec `json:"storage"`
+	GenUUID      *string                  `json:"genUUID,omitempty"`
```

Comment:
This may also be better represented as an annotation which is a similar pattern used elsewhere in Openshift

Reply:
but we need this to be for each node object in this list, otherwise how do we know which node definition is for which deployment/statefulset when later resolving updates

Reply:
Right.  We would have to include `Meta` which is an unnecessary abstraction